packages/block-library/src/navigation-link/index.php

```diff
@@ -105,7 +105,10 @@ function block_core_navigation_link_render_submenu_icon() {
  */
 function render_block_core_navigation_link( $attributes, $content, $block ) {
 	// Don't render the block's subtree if it is a draft.
-	if ( isset( $attributes['id'] ) && is_numeric( $attributes['id'] ) ) {
+	if (
+		isset( $attributes['id'] ) && is_numeric( $attributes['id'] ) &&
+		( 'post' === $attributes['type'] || 'page' === $attributes['type'] )
```

Comment:
I wonder here if we should exclude what `get_post ` does not support (tags, categories) because CPT's (e.g. a `book`) should also follow this draft logic but we can't add them all to this condition.

Reply:
As long as this remains on the radar, I am merging this 🚀